__tests__/app.test.js

```diff
@@ -148,3 +148,61 @@ describe("GET /api/reviews", () => {
 			});
 	});
 });
+
+describe("POST /api/reviews/:review_id/comments. accepts an obj with username and body properties", () => {
+	test("POST 201 adds comment to database, returns comment obj", () => {
+		const testComment = {
+			username: "bainesface",
+			body: "this review is so pointless!",
+		};
+		return request(app)
+			.post("/api/reviews/1/comments")
+			.send(testComment)
+			.expect(201)
+			.then(({ body: { comment } }) => {
+				expect(comment.body).toBe("this review is so pointless!");
+				expect(comment.votes).toBe(0);
+				expect(comment.review_id).toBe(1);
+				expect(comment.comment_id).toBe(7);
+				expect(comment.author).toBe("bainesface");
+				expect(typeof comment.created_at).toBe("string");
+			});
+	});
+	test('POST 404- "sorry, review_id not found!"', () => {
+		const testComment = {
+			username: "bainesface",
+			body: "this review is so pointless!",
+		};
+		return request(app)
+			.post("/api/reviews/20/comments")
+			.send(testComment)
+			.expect(404)
+			.then((response) => {
+				expect(response.body.msg).toBe("sorry, review_id not found!");
+			});
+	});
+	test("POST 400- bad request!", () => {
+		const testComment = {
+			username: "bainesface",
+			body: "this review is so pointless!",
+		};
+		return request(app)
+			.post("/api/reviews/nonsense/comments")
+			.send(testComment)
+			.expect(400)
+			.then((response) => {
+				expect(response.body.msg).toBe("bad request!");
+			});
+	});
+	test("POST 400- Not Acceptable!", () => {
```

Comment:
To confirm these are the tests you will need

Status 201, created comment object
Status 201, ignores unnecessary properties
Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999
Status 400, missing required field(s), e.g. no username or body properties
Status 404, username does not exist

Reply:
i have done this